src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -108,12 +110,16 @@ public AzureAppConfigurationProvider(IConfigurationClientManager configClientMan
             _options = options ?? throw new ArgumentNullException(nameof(options));
             _optional = optional;
 
-            IEnumerable<KeyValueWatcher> watchers = options.ChangeWatchers.Union(options.MultiKeyWatchers);
+            IEnumerable<KeyValueWatcher> watchers = options.KvWatchers.Union(options.FeatureFlagWatchers);
 
             if (watchers.Any())
             {
                 MinRefreshInterval = watchers.Min(w => w.RefreshInterval);
             }
+            else if (options.RegisterAllEnabled)
+            {
+                MinRefreshInterval = options.KvCollectionRefreshInterval;
```

Comment:
if `KvCollectionRefreshInterval` is less than the minimum `FeatureFlagWatchers.RefreshInterval`, MinRefreshInterval will be incorrect